contracts/utils/SignatureValidator.sol

```diff
@@ -99,4 +100,38 @@ contract SignatureValidator {
 
     return signer;
   }
+
+ /**
+   * @notice Returns true if the provided signature is valid for the given signer.
+   * @dev Supports SignatureType.EIP712, SignatureType.EthSign, and ERC1271 signatures
+   * @param _hash      Hash that was signed
+   * @param _signer    Address of the signer candidate
+   * @param _signature Signature byte array
+   */
+  function isValidSignature(
+    bytes32 _hash,
+    address _signer,
+    bytes memory _signature
+  ) internal view returns (bool valid) {
+    uint256 signatureType = uint8(_signature[_signature.length - 1]);
+
+    if (signatureType == SIG_TYPE_EIP712 || signatureType == SIG_TYPE_ETH_SIGN) {
+      // Recover signer and compare with provided
+      valid = recoverSigner(_hash, _signature) == _signer;
+
+    } else if (signatureType == SIG_TYPE_WALLET_BYTES) {
+      // Remove signature type before calling ERC1271, restore after call
+      uint256 prevSize; assembly { prevSize := mload(_signature) mstore(_signature, sub(prevSize, 1)) }
+      valid = ERC1271_MAGICVALUE_BYTES32 == IERC1271Wallet(_signer).isValidSignature(_hash, _signature);
+      assembly { mstore(_signature, prevSize) }
```

Comment:
Yes, it's because modifying the input argument could cause problems outside this context.

We don't do a copy of the `_signature` here, neither does solidity when calling `isValidSignagure`

Reply:
Ok cool and why not make a copy and modify that copy instead? Would be 1 less step

Reply:
Making a copy of the signature would increase the gas cost of the contract